### PR TITLE
ISSUE-986 Fixup guice binding EventDeadLetter for AUTOCOMPLETE EventBus

### PR DESCRIPTION
**Pull request by vttranlina**
resolve: https://github.com/linagora/james-project-private/issues/788

Merge requirement: [james-project] https://github.com/apache/james-project/pull/2205

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.google.inject.name.Named;
+import com.linagora.tmail.james.jmap.EmailAddressContactInjectKeys;
+import com.linagora.tmail.james.jmap.contact.TmailJmapEventSerializer;
+
+public class DistributedEmailAddressContactEventDeadLettersModule extends AbstractModule {
+
+    @Provides
+    @Singleton
+    @Named(EmailAddressContactInjectKeys.AUTOCOMPLETE)
+    EventDeadLetters provideEventDeadLetters(CassandraEventDeadLettersGroupDAO cassandraEventDeadLettersGroupDAO,
```
Comment (Arsnael) [inline tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:22]: Why not adding this block in `DistributedEmailAddressContactEventModule` directly? Do we really need to create an extra module just for that?

**Comment by Arsnael** [comment]
```
[ERROR] /home/jenkins/build/workspace/Tmail_build_PR-1016/tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:[25,46] CassandraEventDeadLettersDAO(com.datastax.oss.driver.api.core.CqlSession,org.apache.james.events.EventSerializer) is not public in org.apache.james.events.CassandraEventDeadLettersDAO; cannot be accessed from outside package
[ERROR] /home/jenkins/build/workspace/Tmail_build_PR-1016/tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:[25,16] CassandraEventDeadLetters(org.apache.james.events.CassandraEventDeadLettersDAO,org.apache.james.events.CassandraEventDeadLettersGroupDAO) is not public in org.apache.james.events.CassandraEventDeadLetters; cannot be accessed from outside package
```

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:22]: I rebased master yesterday :(
Can we do it next time?

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:22]: DistributedEmailAddressContactEventModule is used in both cassandra and postgres
We  should not fixed Cassandra for `DistributedEmailAddressContactEventModule`

**Comment by vttranlina** [comment]
> Merge requirement: [james-project] [apache/james-project#2205](https://github.com/apache/james-project/pull/2205)

> Merge requirement: [james-project] https://github.com/apache/james-project/pull/2205

**Comment by vttranlina** [comment]
> 
> (don't mind the build status, I restarted it by mistake)

> erge requirement: [james-project] https://github.com/apache/james-project/pull/2205

**Comment by Arsnael** [comment]
> erge requirement: [james-project] https://github.com/apache/james-project/pull/2205

It has been merged? :)

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedEmailAddressContactEventDeadLettersModule.java:22]: yes,
my meaning is should not import `Cassandra***` in this class

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**